receiver/stanzareceiver/factory.go

```diff
@@ -17,66 +17,59 @@ package stanzareceiver
 import (
 	"context"
 
-	stanza "github.com/open-telemetry/opentelemetry-log-collection/agent"
+	"github.com/open-telemetry/opentelemetry-log-collection/agent"
+	"github.com/open-telemetry/opentelemetry-log-collection/pipeline"
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/config/configmodels"
 	"go.opentelemetry.io/collector/consumer"
 	"go.opentelemetry.io/collector/receiver/receiverhelper"
 )
 
-const (
-	typeStr = "stanza"
-	verStr  = "0.12.0"
-)
-
-// NewFactory creates a factory for Stanza receiver
-func NewFactory() component.ReceiverFactory {
-	return receiverhelper.NewFactory(
-		typeStr,
-		createDefaultConfig,
-		receiverhelper.WithLogs(createLogsReceiver))
+// LogReceiverType is the interface used by stanza-based log receivers
+type LogReceiverType interface {
+	configmodels.Receiver
+	Version() string
+	CreateDefaultConfig() configmodels.Receiver
+	Decode(configmodels.Receiver) (pipeline.Config, error)
```

Comment:
Maybe better:
```suggestion
	ConfigToOperators(configmodels.Receiver) (pipeline.Config, error)
```

Reply:
I will rename this in the next PR, since I expect the scope of responsibility will change.